pkg/provider/azure/keyvault/keyvault.go

```diff
@@ -671,16 +672,41 @@ func (a *Azure) GetAllSecrets(ctx context.Context, ref esv1.ExternalSecretFind)
 	return secretsMap, nil
 }
 
+func getSecretAllMetadata(tags map[string]*string, attributes *keyvault.Attributes) map[string]*string {
+	metadata := make(map[string]*string)
+	for k, v := range tags {
+		metadata[k] = v
+	}
+	if attributes != nil {
+		for _, field := range reflect.VisibleFields(reflect.TypeOf(*attributes)) {
```

Comment:
What in the name of the almighty coding gods is this doing and why? :D

Reply:
@Skarlso XD I am not sure if its the right approach though.. the basic idea was that there is the KeyAttributes struct which contains the expiry and some other fields.. one of the ways would be to manually add them into the metadata map.. I thought of using reflect for that.. so basically loop over all the exported fields and get the value of that.

Reply:
I'll take a look and let's see if I can unravel something here. 😀